File: Discord_android/Discord/app/src/main/java/com/peanut/discord/models/Channel.java
package com.peanut.discord.models;
import java.util.List;
public class Channel {
    private int channelId;
    private String channelName;
    private int serverId;
    private Server server;
    private List<Message> messages;
    private List<ChannelPermission> channelPermissions;
    public int getChannelId() {
        return channelId;
    }
    public void setChannelId(int channelId) {
        this.channelId = channelId;
    }
    public String getChannelName() {
        return channelName;
    }
    public void setChannelName(String channelName) {
        this.channelName = channelName;
    }
    public int getServerId() {
        return serverId;
    }
    public void setServerId(int serverId) {
        this.serverId = serverId;
    }
    public Server getServer() {
        return server;
    }
    public void setServer(Server server) {
        this.server = server;
    }
    public List<Message> getMessages() {
        return messages;
    }
    public void setMessages(List<Message> messages) {
        this.messages = messages;
    }
    public List<ChannelPermission> getChannelPermissions() {
        return channelPermissions;
    }
    public void setChannelPermissions(List<ChannelPermission> channelPermissions) {
        this.channelPermissions = channelPermissions;
    }
}

File: Discord_win/Discord/Models/Channel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Discord.Models {
    public class Channel {
        public int ChannelId { get; set; }
        public string ChannelName { get; set; }
        public int ServerId { get; set; }

        public Server Server { get; set; }
        public ICollection<Message> Messages { get; set; }
        public ICollection<ChannelPermission> ChannelPermissions { get; set; }
        public void UpdateInfo(Channel source) {
            ChannelName = source.ChannelName;
        }
        public bool SameAs(Channel channel) {
            return ChannelId == channel.ChannelId;
        }
        public Channel Clone() {
            return MemberwiseClone() as Channel;
        }
    }
}
